Formula/solana.rb

```diff
@@ -0,0 +1,34 @@
+class Solana < Formula
+  desc "Web-Scale Blockchain for decentralized apps and marketplaces"
+  homepage "https://solana.com"
+  url "https://github.com/solana-labs/solana/archive/refs/tags/v1.7.11.tar.gz"
+  sha256 "ed70aaed16910eab637eff0499c6e4762350c4ad50340fcb1b4bf94d23f1f97a"
+  license "Apache-2.0"
+
+  depends_on "protobuf" => :build
+  depends_on "rust" => :build
+
+  def install
+    bins = %w[
+      cli
+      bench-streamer
+      faucet
+      keygen
+      log-analyzer
+      net-shaper
+      stake-accounts
+      sys-tuner
+      tokens
+      watchtower
+    ]
+    bins.each do |bin|
+      cd bin do
+        system "cargo", "install", "--no-default-features", *std_cargo_args
+      end
+    end
+  end
+
+  test do
+    system "#{bin}/solana-keygen", "new", "--no-bip39-passphrase", "--no-outfile"
```

Comment:
Can we check the output here?

Reply:
@dawidd6 Sorry, I'm quite new to Ruby and Homebrew and not sure how to do this. The program generates a random key with output as follows (seed phrase redacted). It should be enough to check for exit-code zero.

```
Generating a new keypair
=========================================================================
pubkey: 9VYJ5yPNcd6ePAAXM4MerdA1xbJH4vPncpDmPw3pMfEQ
=========================================================================
Save this seed phrase to recover your new keypair:
elbow lucky elbow lucky elbow lucky elbow lucky elbow lucky elbow lucky
=========================================================================
```

Perhaps we should check for the existence of the following binaries though:

```
ln -s ../Cellar/solana/1.7.11/bin/solana solana
ln -s ../Cellar/solana/1.7.11/bin/solana-bench-streamer solana-bench-streamer
ln -s ../Cellar/solana/1.7.11/bin/solana-faucet solana-faucet
ln -s ../Cellar/solana/1.7.11/bin/solana-keygen solana-keygen
ln -s ../Cellar/solana/1.7.11/bin/solana-log-analyzer solana-log-analyzer
ln -s ../Cellar/solana/1.7.11/bin/solana-net-shaper solana-net-shaper
ln -s ../Cellar/solana/1.7.11/bin/solana-stake-accounts solana-stake-accounts
ln -s ../Cellar/solana/1.7.11/bin/solana-sys-tuner solana-sys-tuner
ln -s ../Cellar/solana/1.7.11/bin/solana-tokens solana-tokens
ln -s ../Cellar/solana/1.7.11/bin/solana-watchtower solana-watchtower
```

Reply:
Checking for the existence of binaries isn't very useful. We can do something like
```suggestion
    assert_match(/pubkey: \w{44}/, shell_output("#{bin}/solana-keygen new --no-bip39-passphrase --no-outfile"))
```